### DISCUSSION TITLE
is there a way to convert extract_all_objects() to file or string of file path

### DISCUSSION THREAD

**Discussion by dwrdx** [mesonbuild/meson#9749] [Q&A]
I'm working on a embedded project and doing cross compile, I wanted to use a program to parse object files that are compiled from .c files before linking them.  

At first, I created a static_library (tmplib) that takes all the source code and archive them to library, then I wrote a custom_target to do the parsing part, for the input argument of this custom_target, I put  tmplib.extract_all_objects().  I got warnings saying extract_all_objects() is not file or str list which  custom_target requires.

I'm wondering if there is a way to workaround. 

Thanks.

**Comment by eli-schwartz** [comment]
I *think* this may just work automatically, if you add them directly to the `command: [...]` array.

**Comment by dwrdx** [comment]
Hi @eli-schwartz , 

I put ```tmplib.extract_all_objects()``` in ```command: [...]``` of my custom_target, but I got following error:
```
Argument <ExtractedObjects 'tmplib.a': <File: hal/hal.c (not built)>, <File: hal/main.c (not built)>]> in "command" is invalid
```
